### PR TITLE
Migrate `run-make/rustdoc-with-short-out-dir-option` to `rmake.rs`

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Part of https://github.com/rust-lang/rust/issues/121876.

r? @jieyouxu

### CODE REVIEW

File: tests/run-make/rustdoc-with-short-out-dir-option/rmake.rs [hunk 1]
```diff
@@ -0,0 +1,16 @@
+use run_make_support::{htmldocck, rustdoc, tmp_dir};
+
+fn main() {
+    let out_dir = tmp_dir().join("rustdoc");
+
+    rustdoc()
+        .input("src/lib.rs")
+        .crate_name("foobar")
+        .crate_type("lib")
+        // This is voluntarily using `-o` option flag and not the `output()` method.
```
Comment (fbstj) [inline tests/run-make/rustdoc-with-short-out-dir-option/rmake.rs:10]:
```suggestion
        // This is intentionally using `-o` option flag and not the `output()` method.
```

Comment (GuillaumeGomez) [reply tests/run-make/rustdoc-with-short-out-dir-option/rmake.rs]: :+1:

**Comment by jieyouxu** [comment]
Thanks, r=me after CI is green

**Pull request merged**